### PR TITLE
Split up `ConfirmationTarget` even more

### PR DESCRIPTION
**Pull request by TheBlueMatt**
This morning we had a sudden feerate spike where feerates spiked something like 100x between one or two blocks, causing my node (and a few other LDK users) to have a feerate estimate 100x lower than that of some of my peers. This cause the new "feerate expansion is treated as dust" logic to kick in and FC two of my channels.

Here we fix this issue by giving the "feerate expansion is treated as dust" logic its own `ConfirmationTarget` which can be yet more conservative so that we're always more conservative than our peers.

Further, because I FC'd while fees were incredibly high, I ended up spending nontrivial funds on the FC transactions, even though there was no particular need to rush - neither of the lost channels had HTLCs so they weren't trying to get confirmed immediately.

Here we fix this second issue by breaking up `OnChainSweep` into an urgent and non-urgent version.

### CODE REVIEW

**Comment by jkczyz** [comment]
cc: @Beige-Coffee

File: lightning/src/chain/chaininterface.rs [hunk 1]
```diff
@@ -49,6 +49,12 @@ pub trait BroadcasterInterface {
 /// estimation.
 #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
 pub enum ConfirmationTarget {
+	/// The most conservative feerate estimate available.
```
[the remark this answers is not in the document]
Comment (valentinewallace) [reply lightning/src/chain/chaininterface.rs]: Would `MaximumAcceptableFee` be accurate/clearer? I agree that conservative could be read in the opposite way...

[the remark this answers is not in the document]
Comment (valentinewallace) [reply lightning/src/chain/chaininterface.rs]: That seems reasonable!

**Comment by valentinewallace** [comment]
CI is sad

File: lightning/src/chain/channelmonitor.rs [hunk 2]
```diff
@@ -2684,6 +2686,30 @@ pub fn deliberately_bogus_accepted_htlc_witness() -> Vec<Vec<u8>> {
...
+			return ConfirmationTarget::UrgentOnChainSweep;
+		}
+		if let Some(txid) = self.current_counterparty_commitment_txid {
+			if !self.counterparty_claimable_outpoints.get(&txid).unwrap().is_empty() {
+				return ConfirmationTarget::UrgentOnChainSweep;
+			}
+		}
+		if let Some(txid) = self.prev_counterparty_commitment_txid {
+			if !self.counterparty_claimable_outpoints.get(&txid).unwrap().is_empty() {
+				return ConfirmationTarget::UrgentOnChainSweep;
```
[the remark this answers is not in the document]
Comment (valentinewallace) [reply lightning/src/chain/channelmonitor.rs:2707]: Btw, there still isn't test coverage here (won't hold the PR up over it though).

**Review by valentinewallace — APPROVED** [review]

**Comment by valentinewallace** [comment]
Benchmark is failing

**Review by valentinewallace — APPROVED** [review]

**Pull request merged**